specs/ContextMenuRequested.md

```diff
@@ -0,0 +1,764 @@
+                if (kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SEPARATOR)
+                {
+                    AppendMenu(hPopupMenu, MF_SEPARATOR, 0, nullptr);
+                }
+                else if (kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_SUBMENU)
+                {
+                    HMENU newMenu = CreateMenu();
+                    wil::com_ptr<ICoreWebView2ContextMenuItemCollection> submenuItems;
+                    CHECK_FAILURE(current->get_Children(&submenuItems));
+                    AddMenuItems(newMenu, submenuItems);
+                    AppendMenu(hPopupMenu, MF_POPUP, (UINT_PTR)newMenu, labelString.c_str());
+                }
+                else if (kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_NORMAL)
+                {
+                    if (isEnabled)
+                    {
+                        AppendMenu(hPopupMenu, MF_BYPOSITION | MF_STRING, commandId, labelString.c_str());
+                    }
+                    else
+                    {
+                        AppendMenu(hPopupMenu, MF_GRAYED | MF_STRING, commandId, labelString.c_str());
+                    }
+                }
+                else if (
+                    kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_CHECKBOX ||
+                    kind == COREWEBVIEW2_CONTEXT_MENU_ITEM_KIND_RADIO)
+                {
+                    if (isEnabled)
+                    {
+                        if (isChecked)
+                        {
+                            AppendMenu(hPopupMenu, MF_CHECKED | MF_STRING, commandId, labelString.c_str());
+                        }
+                        else
+                        {
+                            AppendMenu(hPopupMenu, MF_BYPOSITION | MF_STRING, commandId, labelString.c_str());
+                        }
+                    }
+                    else
+                    {
+                        if (isChecked)
+                        {
+                            AppendMenu(
+                                hPopupMenu, MF_CHECKED | MF_GRAYED | MF_STRING, commandId, labelString.c_str());
+                        }
+                        else
+                        {
+                            AppendMenu(hPopupMenu, MF_GRAYED | MF_STRING, commandId, labelString.c_str());
+                        }
+                    }
+                }
+            }
+        }
+```
+## .Net/ WinRT Add or Remove Entries From WebView Menu 
+
+ ```c#
+    webView.CoreWebView2.ContextMenuRequested += delegate (object sender, CoreWebView2ContextMenuRequestedEventArgs args)
+    {
+        IList<CoreWebView2ContextMenuItem> menuList = args.MenuItems;
+        CoreWebView2ContextKind context = args.ContextMenuInfo.ContextKind;
+        args.Handled = false;
+        if (context == CoreWebView2ContextKind.Image)
+        {
+            // removes the last item in the collection
+            menuList.RemoveAt(menuList.Count - 1);
```

Comment:
This should align with the C++ version and look for the "Save As" item for removal.

```csharp
for (int index = 0; index < menuList.Count; index++)
{
    if (menuList[index].Descriptor == CoreWebView2ContextMenuItemDescriptor.SaveImageAs)
    {
        menuList.RemoveAt(index);
        break;
    }
}
```